llvm/include/llvm/CodeGen/CodeGenPassBuilder.h

```diff
@@ -290,6 +291,9 @@ template <typename DerivedT> class CodeGenPassBuilder {
                                    inconvertibleErrorCode());
   }
 
+  /// Target can override this to add GlobalMergePass before all IR passes.
+  void addGlobalMergePass(AddIRPass &) const {}
```

Comment:
Why is this here? Is there an analogous hook in the old PM?

Reply:
Something doesn't make sense. Why is this a function pass, that only changes the module and does all the work in doInitalization? Should it be converted to a ModulePass?

Reply:
Seems to be GlobalMerge should be run earlier than it is now. I would expect it to run before CodeGenPrepare at least (if it even belongs in the codegen pipeline in the first place)

Reply:
It sounds like it already is running at the very beginning of the codegen pipeline?  doInitialization runs before any passes, I think.

If it did run in the optimization pipeline, it would have to be very late; the target hooks use information about the exact load/store ops that access the global to pick appropriate thresholds.  And it messes with alias analysis a bit for mutable variables.  So ultimately it wouldn't be much different from running it in early codegen.  (Which is really too early for constants, given the pass can't see constant pools, but that's a different issue.)